tests/drivers/abp2/main.c

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (C) 2024 CNRS, France
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup tests
+ * @{
+ *
+ * @file
+ * @brief       Test application for the ABP2 Honeywell ABP2 series
+ *              pressure and temperature sensor driver.
+ *
+ * @author      David Picard <david.picard@clermont.in2p3.fr>
+ * @}
+ */
+
+#include <stdio.h>
+#include <string.h>
+
+#include "ztimer.h"
+#include "timex.h"
+#include "phydat.h"
+#include "abp2.h"
+#include "abp2_params.h"
+
+#define STATUS_SLEEP_MS         (1)
+#define MEASUREMENT_SLEEP_MS    (1000)
+#define MAX_LOOPS_STATUS        (10)
+#define MAX_LOOPS_MEAS          (5)
+
+static abp2_t dev;
+
+int main(void)
+{
+    int32_t press = 0;
+    int32_t temp = 0;
+    abp2_params_t prms;
+    phydat_t phyPress;
+    phydat_t phyTemp;
+    void *ptr;
+    int blockingMode = 1;
+    int cntMeas = 0;
+
+    puts("ABP2 Honeywell series pressure and temperature sensor\n");
+
+    ptr = memcpy(&prms, &abp2_params, sizeof(abp2_params));
+    if(!ptr)
+    {
+       puts("main() >> ERROR: memcpy() failed");
+       return 1;
+    }
+
+#ifdef BOARD_NUCLEO_L031K6
+    /* change chip select pin: */
+    prms.cs = GPIO_PIN(PORT_A, 11);
+#endif
+
+    printf("Initializing SPI bus %d\n", prms.spi);
+    spi_init(prms.spi);
+
+    printf("Initializing ABP2 at SPI_DEV(%i)... ", prms.spi);
+
+    if (abp2_init(&dev, &prms) == 0)
+    {
+        puts("[OK]");
+    }
+    else {
+        puts("[Failed]");
+        return -1;
+    }
+#ifdef BOARD_NUCLEO_L031K6
+    ztimer_sleep(ZTIMER_MSEC, 5000);
+#endif
```

Comment:
Why does this board need an extra pause?

Reply:
Debug stuff. I removed it.